src/main/java/emu/grasscutter/game/entity/EntityBaseGadget.java

```diff
@@ -73,6 +75,11 @@ protected EntityBaseGadget(Scene scene, CreateGadgetEntityConfig gadgetCreateCon
         this.arguments = gadgetCreateConfig.getArguments();
         this.owner = gadgetCreateConfig.getPlayerOwner();
 
+        GatherData gatherData = GameData.getGatherDataMap().get(gadgetCreateConfig.getPointType());
+        if(gatherData != null) {
+            this.interactEnabled = !gatherData.initDisableInteract();
```

Comment:
This should probably be handled in GadgetGatherPoint when the CreateGadgetEntityConfig is created, instead of it being overwritten here. In the future plugins should be able to modify the CreateConfig, before the entity is created, so this would otherwise overwrite it.

Reply:
So there's planned feature for plugins on creating gadgets. Could you brief me about that planned feature?